sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -878,12 +878,17 @@ case class ShowTablesCommand(
       //
       // Note: tableIdentifierPattern should be non-empty, otherwise a [[ParseException]]
       // should have been thrown by the sql parser.
-      val tableIdent = TableIdentifier(tableIdentifierPattern.get, Some(db))
-      val table = catalog.getTableMetadata(tableIdent).identifier
-      val partition = catalog.getPartition(tableIdent, partitionSpec.get)
-      val database = table.database.getOrElse("")
-      val tableName = table.table
-      val isTemp = catalog.isTemporaryTable(table)
+      val table = catalog.getTableMetadata(TableIdentifier(tableIdentifierPattern.get, Some(db)))
+      val tableIdent = table.identifier
```

Comment:
Is there a reason you change from
```scala
val tableIdent = TableIdentifier(tableIdentifierPattern.get, Some(db))
val table = catalog.getTableMetadata(tableIdent).identifier
```
to 
```scala
val table = catalog.getTableMetadata(TableIdentifier(tableIdentifierPattern.get, Some(db)))
val tableIdent = table.identifier
```

Reply:
If there is a difference, I'm wondering if we have a test coverage for this change, @MaxGekk .